source/index.ts

```diff
@@ -60,6 +60,8 @@ export default class Conf<T extends Record<string, any> = Record<string, unknown
 	readonly #encryptionKey?: string | Buffer | NodeJS.TypedArray | DataView;
 	readonly #options: Readonly<Partial<Options<T>>>;
 	readonly #defaultValues: Partial<T> = {};
+	#readFromDisk = true;
+	#state?: string;
```

Comment:
I support the PR, but I'd like to suggest a few improvements, if you don't mind.

1. To avoid breaking change introduce new option like `memoize/useMemoryState`
2. 
```ts
get store(): T {
		const state = createPlainObject();

		if (this.options.memoize) {
		   if (this.#state) {
		      return this.#state;
		   }

		   this.#state = state;
		}
		
		try {
			const data = fs.readFileSync(this.path, this.#encryptionKey ? null : 'utf8');
			const dataString = this._encryptData(data);
			const deserializedData = this._deserialize(dataString);
			this._validate(deserializedData);
			
			return Object.assign(state, deserializedData);
		} catch (error: unknown) {
			if ((error as any)?.code === 'ENOENT') {
				this._ensureDirectory();
				return state;
			}
			if (this.#options.clearInvalidConfig && (error as Error).name === 'SyntaxError') {
				return state;
			}
			throw error;
		}
	}
```
3. Use `change` event to trigger `state` reset instead of `_watch`
```ts
if (options.memoize) {
  this.events.on('change', this._resetState)
}
// ...
_resetState() {
   this.#state = undefined;
}
```

Reply:
@sindresorhus,

Could you plz take a look?

Reply:
I agree with the suggestions here. It should be behind an option. We can make the option true by default at some point when it's more battle-tested.